package fr.neocle.flexbans.velocity.commands;

import com.velocitypowered.api.command.SimpleCommand;
import com.velocitypowered.api.proxy.ProxyServer;
import fr.neocle.flexbans.common.commands.punishments.kick.KickCommandExecutor;
import fr. neocle.flexbans.commands.punishments.kick.KickExecutor;
import fr.neocle.flexbans.velocity.commands.adapters.VelocityCommandInvocation;
import fr.neocle.flexbans.velocity.commands.helpers.punishments.VelocityKickCommandHelper;

import java.util.List;

public class KickCommand implements SimpleCommand {
    private final KickCommandExecutor commandExecutor;

    public KickCommand(KickExecutor kickExecutor, ProxyServer proxyServer) {
        this.commandExecutor = new KickCommandExecutor(
                kickExecutor,
                new VelocityKickCommandHelper(proxyServer)
        );
    }

    @Override
    public void execute(Invocation invocation) {
        var wrappedInvocation = new VelocityCommandInvocation(invocation, invocation.source());
        commandExecutor.execute(wrappedInvocation);
    }

    @Override
    public List<String> suggest(Invocation invocation) {
        var wrappedInvocation = new VelocityCommandInvocation(invocation, invocation.source());
        return commandExecutor. suggest(wrappedInvocation);
    }

    @Override
    public boolean hasPermission(Invocation invocation) {
        return invocation.source().hasPermission("flexbans.kick");
    }
}